File: 0110-balanced-binary-tree/0110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.ans = True
    def isB(self, root: Optional[TreeNode]) -> int:
        if root == None:
            return 0
        left = self.isB(root.left)
        right = self.isB(root.right)
        if(abs(left-right) > 1):
            self.ans = False
        return 1 + max(left,right)
        
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if root == None:
            return True
        val = self.isB(root)
        return self.ans
        

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.cpp
class Solution {
private:
    int check(TreeNode* root)
    {
        if(root == NULL) return 0 ;

        int left = check(root->left);
        int right = check(root->right);
        if(left == -1 || right == -1||abs(left-right)>1) return -1;
        return 1 + max(left,right);
    }
public:
    bool isBalanced(TreeNode* root) {
        int heigthCheck = check(root);
        return heigthCheck!=-1;
    }
};